Sources/SourceKit/SourceKitServer.swift

```diff
@@ -468,6 +470,57 @@ extension SourceKitServer {
     }
   }
 
+  // FIXME: a lot of duplication with definition request
+  func implementation(_ req: Request<ImplementationRequest>, workspace: Workspace) {
+    // FIXME: sending yourself a request isn't very convenient
+
+    guard let service = workspace.documentService[req.params.textDocument.url] else {
+      req.reply([])
+      return
+    }
+
+    let id = service.send(SymbolInfoRequest(textDocument: req.params.textDocument, position: req.params.position), queue: queue) { result in
+      guard let symbols: [SymbolDetails] = result.success ?? nil, let symbol = symbols.first else {
+        if let error = result.failure {
+          req.reply(.failure(error))
+        } else {
+          req.reply([])
+        }
+        return
+      }
+
+      guard let usr = symbol.usr, let index = workspace.index else {
+        return req.reply([])
+      }
+    
+      var occurs = index.occurrences(ofUSR: usr, roles: .baseOf)
+      if occurs.isEmpty {
+        occurs = index.occurrences(relatedToUSR: usr, roles: .overrideOf)
+      }
+
+      // FIXME: overrided method logic
```

Comment:
What is this FIXME about?

Reply:
Just got copy-pasted with the rest of `definition(_:workspace:)`

Reply:
I don't think it's relevant here.

Reply:
deleted